### PR TITLE
adding once feature

### PR DESCRIPTION
**Pull request by iruzevic**
### Added

- New feature to allow the form to be submitted only once per user.
- New feature to allow the form to be visible only when the user is logged in.

### CODE REVIEW

File: src/Blocks/custom/forms/forms.php [hunk 1]
```diff
@@ -16,6 +16,11 @@
 $manifestInvalid = Helpers::getComponent('invalid');
 $manifestSettings = Helpers::getSettings();
 
+// Check is there any reason to not render forms block.
```
Comment (piqusy) [inline src/Blocks/custom/forms/forms.php:19]:
```suggestion
// Check if is there any reason to not render forms block.
```

File: src/Validation/SettingsValidation.php [hunk 2]
```diff
@@ -208,6 +220,46 @@ public function getSettingsGlobalData(): array
...
+						'tabLabel' => \__('Users', 'eightshift-forms'),
+						'tabContent' => [
+							[
+								'component' => 'checkboxes',
+								'checkboxesFieldLabel' => '',
+								'checkboxesName' => UtilsSettingsHelper::getOptionName(self::SETTINGS_VALIDATION_USE_ONLY_LOGGED_IN_KEY),
+								'checkboxesContent' => [
+									[
+										'component' => 'checkbox',
+										'checkboxLabel' => \__('Use only for logged in users', 'eightshift-forms'),
```
Comment (piqusy) [inline src/Validation/SettingsValidation.php:234]:
```suggestion
										'checkboxLabel' => \__('Show only to logged in users', 'eightshift-forms'),
```

File: src/Validation/SettingsValidation.php [hunk 3]
```diff
@@ -208,6 +220,46 @@ public function getSettingsGlobalData(): array
...
+							...($loggedInForm ? [
+								[
+									'component' => 'checkboxes',
+									'checkboxesFieldLabel' => '',
+									'checkboxesName' => UtilsSettingsHelper::getOptionName(self::SETTINGS_VALIDATION_USE_SUBMIT_ONCE_KEY),
+									'checkboxesContent' => [
+										[
+											'component' => 'checkbox',
+											'checkboxLabel' => \__('Use single submit per user', 'eightshift-forms'),
+											'checkboxHelp' => \__('Use only single submit per user.', 'eightshift-forms'),
```
Comment (piqusy) [inline src/Validation/SettingsValidation.php:252]:
```suggestion
											'checkboxHelp' => \__('If enabled, each user can submit the form only once.', 'eightshift-forms'),
```

**Review by piqusy — CHANGES_REQUESTED** [review]
Nice work 👏🏼 

Left a few wording suggestions

File: src/Blocks/custom/forms/forms.php (hunk 1 above)
Comment (kancijan) [reply src/Blocks/custom/forms/forms.php:19]:
```suggestion
// Check if there is any reason not to render forms block.
```

File: src/Validation/SettingsValidation.php [hunk 4]
```diff
@@ -208,6 +220,46 @@ public function getSettingsGlobalData(): array
...
+						'tabContent' => [
+							[
+								'component' => 'checkboxes',
+								'checkboxesFieldLabel' => '',
+								'checkboxesName' => UtilsSettingsHelper::getOptionName(self::SETTINGS_VALIDATION_USE_ONLY_LOGGED_IN_KEY),
+								'checkboxesContent' => [
+									[
+										'component' => 'checkbox',
+										'checkboxLabel' => \__('Use only for logged in users', 'eightshift-forms'),
+										'checkboxHelp' => \__('If form is available only for the logged in users. If the user is not logged in the form will not show.', 'eightshift-forms'),
```
Comment (kancijan) [inline src/Validation/SettingsValidation.php:235]:
```suggestion
										'checkboxHelp' => \__('The form will be accessible only to users who are logged in.', 'eightshift-forms'),
```

**Review by kancijan — DISMISSED** [review]

**Review by kancijan — APPROVED** [review]

**Review by goranalkovic-infinum — APPROVED** [review]

**Pull request merged**